core/src/main/java/org/infinispan/configuration/cache/PartitionHandlingConfiguration.java

```diff
@@ -19,10 +19,12 @@
          .build();
    public static final AttributeDefinition<PartitionHandling> WHEN_SPLIT = AttributeDefinition.builder("whenSplit", PartitionHandling.ALLOW_READ_WRITES)
          .immutable().build();
-   public static final AttributeDefinition<EntryMergePolicy> MERGE_POLICY = AttributeDefinition.builder("mergePolicy", MergePolicies.PREFERRED_ALWAYS).immutable().build();
+   public static final AttributeDefinition<EntryMergePolicy> MERGE_POLICY = AttributeDefinition.builder("mergePolicy", MergePolicy.PREFERRED_ALWAYS.getImpl(), EntryMergePolicy.class).immutable().build();
+   public static final AttributeDefinition<String> MERGE_POLICY_CLASS = AttributeDefinition.builder("mergePolicyClass",
```

Comment:
I don't think this should be a string, but an actual class. The XML parser can perform the Class.forName stuff, but in programmatic config I wouldn't want to use the name.

Reply:
@tristantarrant @ryanemerson I don't think we allow the user to pass in a class anywhere else in the programmatic configuration. Usually it's the parser's job to create an instance of the class in the XML configuration with `Util.getInstance(className, classLoader)`.

Reply:
No, we pass instances usually (e.g. Transport)

Reply:
@tristantarrant I'm not sure using Class is applicable here. I'm using a string as I am following a similar deployment method as used by the `DeployedCacheStoreFactory`. So the problem is that the Class may not be present on the classpath when a cache config is created, but instead will be loaded once the policy has been deployed and the cache startsup.

Reply:
Actually, thinking about this some more. Another option is to leave the standard `PartitonHandlingConfiguration` as before this PR and create `ServerPartitionHandlingConfiguration extends PartitionHandlingConfiugration` class purely for use in the server. This would allow a class name to be specified as a String internally without polluting `PartitionHandlingConfiguration`.